src/store/modules/permissions.js

```diff
@@ -37,18 +46,63 @@ export const checkPermissions = async (method, params = {}) => {
   return limitLeft === 0;
 };
 
+export const requestAddressForAepp = async (aepp, address, connectionPopup) => {
+  const { permissions } = await getState();
+  const { host } = getAeppUrl(aepp);
+  const permissionsHost = permissions?.[host];
+  if (permissionsHost?.addresses.includes(address)) return true;
+
+  try {
+    if (await checkPermissions(host, { method: 'connection.open' })) await connectionPopup();
+    if (!permissionsHost) await addNewHost(host, address);
+  } catch {
+    return false;
+  }
+
+  return true;
+};
+
 export default {
   namespaced: true,
 
```

Comment:
```suggestion

  state: {
    'superhero.com': {
      // allow connection and transaction/message signing without confirmation
      // not mentioning a particular address
    },
    ...{ 'localhost:8080': ... }
  },
```
can we instead of `setNewHosts` do like this?